File: parse_seo.js
const fs = require('fs');

function parseSeo(filename) {
    const html = fs.readFileSync(filename, 'utf-8');
    
    const getTagContent = (regex) => {
        const match = html.match(regex);
        return match ? match[1].trim() : "NONE";
    };

    const title = getTagContent(/<title[^>]*>([^<]+)<\/title>/i);
    const desc = getTagContent(/<meta[^>]+name="description"[^>]+content="([^"]+)"/i) 
                 || getTagContent(/<meta[^>]+content="([^"]+)"[^>]+name="description"/i);
    const canonical = getTagContent(/<link[^>]+rel="canonical"[^>]+href="([^"]+)"/i) 
                      || getTagContent(/<link[^>]+href="([^"]+)"[^>]+rel="canonical"/i);
    const robots = getTagContent(/<meta[^>]+name="robots"[^>]+content="([^"]+)"/i) 
                   || getTagContent(/<meta[^>]+content="([^"]+)"[^>]+name="robots"/i);
    
    // OG Tags
    const ogUrl = getTagContent(/<meta[^>]+property="og:url"[^>]+content="([^"]+)"/i) 
                  || getTagContent(/<meta[^>]+content="([^"]+)"[^>]+property="og:url"/i);
    const ogImage = getTagContent(/<meta[^>]+property="og:image"[^>]+content="([^"]+)"/i) 
                    || getTagContent(/<meta[^>]+content="([^"]+)"[^>]+property="og:image"/i);
                    
    // H1 count
    const h1s = [...html.matchAll(/<h1[^>]*>/gi)];
    const h1Count = h1s.length;

    // JSON-LD
    const jsonLd = html.includes('application/ld+json') ? "YES" : "NO";

    console.log(`\n--- ${filename} ---`);
    console.log(`Title: ${title}`);
    console.log(`Description: ${desc}`);
    console.log(`Canonical: ${canonical}`);
    console.log(`Robots: ${robots}`);
    console.log(`H1 Count: ${h1Count}`);
    console.log(`OG:URL: ${ogUrl}`);
    console.log(`OG:Image: ${ogImage}`);
    console.log(`JSON-LD: ${jsonLd}`);
}

const files = process.argv.slice(2);
files.forEach(f => parseSeo(f));

File: parse_seo.py
import sys
from html.parser import HTMLParser

class SEOParser(HTMLParser):
    def __init__(self):
        super().__init__()
        self.title = ""
        self.in_title = False
        self.meta_desc = ""
        self.canonical = ""
        self.robots = ""
        self.h1_count = 0
        self.og_url = ""
        self.og_image = ""
        self.json_ld = False

    def handle_starttag(self, tag, attrs):
        attrs_dict = dict(attrs)
        if tag == "title":
            self.in_title = True
        elif tag == "meta":
            if attrs_dict.get("name") == "description":
                self.meta_desc = attrs_dict.get("content", "")
            elif attrs_dict.get("name") == "robots":
                self.robots = attrs_dict.get("content", "")
            elif attrs_dict.get("property") == "og:url":
                self.og_url = attrs_dict.get("content", "")
            elif attrs_dict.get("property") == "og:image":
                self.og_image = attrs_dict.get("content", "")
        elif tag == "link":
            if attrs_dict.get("rel") == "canonical":
                self.canonical = attrs_dict.get("href", "")
        elif tag == "h1":
            self.h1_count += 1
        elif tag == "script":
            if attrs_dict.get("type") == "application/ld+json":
                self.json_ld = True

    def handle_data(self, data):
        if self.in_title:
            self.title += data

    def handle_endtag(self, tag):
        if tag == "title":
            self.in_title = False

for filename in sys.argv[1:]:
    try:
        with open(filename, "r") as f:
            content = f.read()
            parser = SEOParser()
            parser.feed(content)
            print(f"--- {filename} ---")
            print(f"Title: {parser.title}")
            print(f"Desc: {parser.meta_desc}")
            print(f"Canonical: {parser.canonical}")
            print(f"Robots: {parser.robots}")
            print(f"H1 Count: {parser.h1_count}")
            print(f"OG URL: {parser.og_url}")
            print(f"OG Image: {parser.og_image}")
            print(f"JSON-LD: {parser.json_ld}")
    except Exception as e:
        print(f"Error reading {filename}: {e}")
